Makefile

```diff
@@ -174,34 +202,64 @@ site/content/usage/20-schema.md: $(call godeps,cmd/schema/generate.go)
 $(generated_code_aws_sdk_mocks): $(call godeps,pkg/eks/mocks/mocks.go)
 	mkdir -p vendor/github.com/aws/
 	@# Hack for Mockery to find the dependencies handled by `go mod`
-	ln -sfn "$$(go env GOPATH)/pkg/mod/github.com/aws/aws-sdk-go@v1.23.15" vendor/github.com/aws/aws-sdk-go
+	ln -sfn "$(gopath)/pkg/mod/github.com/aws/aws-sdk-go@v1.23.15" vendor/github.com/aws/aws-sdk-go
 	time env GOBIN=$(GOBIN) go generate ./pkg/eks/mocks
 
 ##@ Docker
 
-ifeq ($(OS),Windows_NT)
-TEST_TARGET=unit-test
-else
-TEST_TARGET=test
-endif
-
-.PHONY: eksctl-deps-image
-eksctl-deps-image: ## Create a cache image with dependencies
-	-time docker pull $(EKSCTL_DEPENDENCIES_IMAGE)
-	@# Pin dependency file permissions.
-	@# Docker uses the file permissions as part of the COPY hash, which can lead to cache misses
-	@# in hosts with different default file permissions (umask).
-	chmod 0700 install-build-deps.sh
-	chmod 0600 go.mod go.sum
-	time docker build --cache-from=$(EKSCTL_DEPENDENCIES_IMAGE) --tag=$(EKSCTL_DEPENDENCIES_IMAGE) -f Dockerfile.deps .
+update_build_image_manifest = git ls-tree --full-tree @ -- $(build_image_input) > .build_image_manifest
+
+.PHONY: check-build-image-manifest-up-to-date
+check-build-image-manifest-up-to-date: ## Update build image manifest and commits the changes
+	git diff --quiet -- $(build_image_input) || (git --no-pager diff $(build_image_input); exit 1)
+	$(update_build_image_manifest)
+	git diff --quiet -- .build_image_manifest || (git --no-pager diff .build_image_manifest; echo "HINT: to fix this, run 'make update-build-image-manifest'"; exit 1)
+
+.PHONY: update-build-image-manifest
+update-build-image-manifest: ## Update build image manifest and commits the changes
+	$(update_build_image_manifest)
+	git commit --quiet .build_image_manifest --message 'Update build image manifest'
+	$(info build image name is $(build_image_name))
+
+.PHONY: build-image
+build-image: check-build-image-manifest-up-to-date ## Build the build image that has all of external dependencies
+	-docker pull $(build_image_name)
```

Comment:
I know I'm not on the reviewer's list but looks like an extra `-` here...

Reply:
The list of reviewers is just a suggestion :)

You mean `-docker`? That's make syntax for ignoring errors.